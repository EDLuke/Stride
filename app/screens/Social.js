import React from 'react';
import {Font} from 'expo';
import { Button, FlatList, ScrollView, StyleSheet, Text, View } from 'react-native';

GLOBAL = require('../components/CurrentUser');

// class MyListItem extends React.PureComponent { 
//   _onPress = () => { 
//     this.props.onPressItem(this.props.id); 
//   }; 
//   render() {
//     <Text style={styles.boxtext}
//       onPress={() => this.props.navigation.navigate('SocialDetail')}
//     >{item}</Text>
//     } 
// }

export default class Social extends React.Component{
	state = {
	};

	render() {
		return (
			<View style = {styles.container}>
              <View style={styles.textBoxContainer}>
            		<Text style={styles.title}>Friends' Circle</Text>
              </View>
        		<View style={styles.textBoxContainer}>	
        				<Text style={styles.boxtitle}>LeaderBoard</Text>
            </View>
        			<View style={styles.leaderBoardBoxContainer}>
                  <ScrollView>
                  <FlatList
                    data = {GLOBAL.currentUser.friends}
                    keyExtractor = {(item, index) => item}
                    renderItem = {({item}) => <Text style={styles.boxtext}
                                                onPress={() => this.props.navigation.navigate('SocialDetail', {name: item})}
                                              >{item}</Text>}
                  />
                  </ScrollView>
              </View>
			</View>
      );
	}
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  title: {
    fontFamily: 'Helvetica',
    fontSize: 35,
    fontWeight: 'bold',
    color: '#494949'
  },
  subtitle: {
    fontFamily: 'Helvetica',
    fontSize: 25,
    color: '#686868'
  },
  boxtitle: {
  	fontFamily: 'Helvetica',
    fontSize: 25,
    fontWeight: 'bold',
    color: '#686868'
  },
  boxtext:{
    fontFamily: 'Helvetica',
    fontSize: 20,
    color: '#686868'
  },
  textBoxContainer: {
  	flex: 1,
  	backgroundColor: '#fff',
  	justifyContent: 'flex-start',
    marginTop: 10,
    padding: 10
  },
  leaderBoardBoxContainer: {
    flex: 3, 
  	backgroundColor: '#bbb',
  	justifyContent: 'flex-start',
    aspectRatio: 3

  },
  buttonContainer: {
    margin: 50,
  }
});
