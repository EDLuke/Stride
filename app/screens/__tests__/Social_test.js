import 'react-native';
import React from 'react';
import Social from '../Social';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Social />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});