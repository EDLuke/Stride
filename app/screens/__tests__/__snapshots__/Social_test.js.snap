// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": 10,
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#494949",
          "fontFamily": "Helvetica",
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
      Friends' Circle
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": 10,
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#686868",
          "fontFamily": "Helvetica",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      LeaderBoard
    </Text>
  </View>
  <View
    style={
      Object {
        "aspectRatio": 3,
        "backgroundColor": "#bbb",
        "flex": 3,
        "justifyContent": "flex-start",
      }
    }
  >
    <RCTScrollView>
      <View>
        <RCTScrollView
          data={undefined}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          onViewableItemsChanged={undefined}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
